x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/stats/VerifierMetricsTests.java

```diff
@@ -0,0 +1,209 @@
+    }
+    
+    public void testSequenceWithThreeQueries() {
+        Counters c = eql("sequence with maxspan=1d\r\n" + 
+            "  [process where serial_event_id < 4] by exit_code\r\n" + 
+            "  [process where opcode == 1] by user\r\n" + 
+            "  [file where parent_process_name == \"file_delete_event\"] by exit_code\r\n" + 
+            "| head 4");
+        assertCounters(1L, 0, 0, 1L, 0, 1L, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 0, 1L, 0, 0, 0, 0, c);
+    }
+    
+    public void testSequenceWithFiveQueries() {
+        Counters c = eql("sequence with maxspan=1d\r\n" + 
+            "  [process where serial_event_id < 4] by exit_code\r\n" + 
+            "  [process where opcode == 1] by user\r\n" + 
+            "  [file where parent_process_name == \"file_delete_event\"] by exit_code\r\n" +
+            "  [process where serial_event_id < 4] by exit_code\r\n" + 
+            "  [process where opcode == 1] by user\r\n" + 
+            "| head 4");
+        assertCounters(1L, 0, 0, 1L, 0, 1L, 0, 0, 0, 0, 0, 1L, 0, 0, 0, 0, 1L, 0, 0, 0, 0, c);
+    }
+
+    public void testSequenceWithSevenQueries() {
+        Counters c = eql("sequence by exit_code, user\r\n" + 
+            "  [process where serial_event_id < 4]\r\n" + 
+            "  [process where opcode == 1]\r\n" + 
+            "  [file where parent_process_name == \"file_delete_event\"]\r\n" +
+            "  [process where serial_event_id < 4]\r\n" + 
+            "  [process where opcode == 1]\r\n" + 
+            "  [process where true]\r\n" + 
+            "  [process where true]\r\n" + 
+            "| tail 1");
+        assertCounters(1L, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 1L, 0, 0, 0, c);
+    }
+    
+    public void testSequenceWithThreeKeys() {
+        Counters c = eql("sequence by exit_code, user, serial_event_id\r\n" + 
+            "  [process where serial_event_id < 4]\r\n" + 
+            "  [process where opcode == 1]\r\n");
+        assertCounters(1L, 0, 0, 1L, 0, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1L, 0, 0, c);
+    }
+
+    public void testSequenceWithFourKeys() {
+        Counters c = eql("sequence by exit_code, user, serial_event_id, pid\r\n" + 
+            "  [process where serial_event_id < 4]\r\n" + 
+            "  [process where opcode == 1]\r\n");
+        assertCounters(1L, 0, 0, 1L, 0, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1L, 0, c);
+    }
+
+    public void testSequenceWithFiveKeys() {
+        Counters c = eql("sequence by exit_code, user, serial_event_id, pid, ppid\r\n" + 
+            "  [process where serial_event_id < 4]\r\n" + 
+            "  [process where opcode == 1]\r\n");
+        assertCounters(1L, 0, 0, 1L, 0, 0, 0, 0, 1L, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1L, c);
+    }    
+
+    private void assertCounters(long sequence, long event, long join, long head, long tail, long seqMaxSpan, long seqUntil, long joinUntil,
```

Comment:
It would be better to have an object that contains all the counters so moving them around doesn't affect the method signature and thus all calls of it.

Reply:
+1